src/browser/api/webcontents.ts

```diff
@@ -1,9 +1,63 @@
 import * as url from 'url';
 import { app } from 'electron';
+import { Identity } from '../../shapes';
 import ofEvents from '../of_events';
-import route from '../../common/route';
+import route, { WindowRoute } from '../../common/route';
 import { InjectableContext, EntityType } from '../../shapes';
 
+export function hookWebContentsEvents(webContents: Electron.WebContents, { uuid, name }: Identity, topic: string, routeFunc: WindowRoute) {
+    webContents.on('did-get-response-details', (e,
+        status,
+        newUrl,
+        originalUrl,
+        httpResponseCode,
+        requestMethod,
+        referrer,
+        headers,
+        resourceType
+    ) => {
+        const type = 'resource-response-received';
+
+        const payload = {
+            name,
+            uuid,
+            topic,
+            type,
+            status,
+            newUrl,
+            originalUrl,
+            httpResponseCode,
+            requestMethod,
+            referrer,
+            headers,
+            resourceType
+        };
+        ofEvents.emit(routeFunc(type, uuid, name), payload);
+    });
+    webContents.on('did-fail-load', (e,
+        errorCode,
+        errorDescription,
+        validatedURL,
+        isMainFrame
+    ) => {
+        const type = 'resource-load-failed';
+        const payload = {
+            name,
+            uuid,
+            topic,
+            type,
+            errorCode,
+            errorDescription,
+            validatedURL,
+            isMainFrame
+        };
+        ofEvents.emit(routeFunc(type, uuid, name), payload);
+    });
+    webContents.once('destroyed', () => {
```

Comment:
`close` has been changed to `destroyed`. I'm assuming this is `webContents` specific?

Reply:
yes